### PR TITLE
fix(@desktop/general): biometrics prompt is shown after restarting the app even it was not enabled

### PR DESCRIPTION
**Pull request by saledjenic**
Fixes: #11261

### CODE REVIEW

**Comment by saledjenic** [comment]
Two things that I've noticed:

1. In the past (when I checked that part last time) we had three options when a user wanted to change the biometrics settings, like... store, ask me later, and never stores. Now there are only 2 options (the switch on the settings profile page) store/don't store, but the values for that field remained the same (store, notNow, never). 

Now what does that have as a consequence, if during the onboarding flow, at the last step when the user has to select among option 1 "Yes, use touch ID" and option 2 "I prefer to use my PIN/Password", if the user opt-ins for option 2, we store `never` and in that case we don't ask for biometrics at all and that's ok. But if the user selects option 1, we temporarily store `notNow`, then when a user successfully stores the pass/pin we store the `store` value. Why we're using `notNow` to differentiate stored value from "offer to store". Now what's the issue... if the user during onboarding selects option 1 and later when he gets into the app he will see the store to keychain toggle set to off in the profile settings, what is ok and will be asked to provide the pass for storing to keychain, to complete the action. Now if he discards the OS keychain popup, he will be asked to store pass/pin to the Keychain each time he logs in to the app and the only way to stop that will be to store the pass/pin and then remove it, cause there is no third option on the profile settings page. That can be annoying.

2. Another issue I saw and mentioned it few times a long time ago, ~2 years ago, but it's still here is that previously we've been storing local .ini (settings) files by 3-word names and that for sure was a unique way. Then we switched to storing them by display name. That's misleading for all of them, for all created Status accounts are stored at the same location, which means that having two Status account with the same display name lead to inconsistencies, cause the last one overwrites the previous one.

File: src/app/global/local_account_settings.nim [hunk 1]
```diff
@@ -33,8 +34,15 @@ QtObject:
...
     if(not self.settings.isNil):
       self.settings.delete
-    let filePath = os.joinPath(self.settingsFileDir, fileName)
-    self.settings = newQSettings(filePath, QSettingsFormat.IniFormat)
+    let
+      currentFilePath = os.joinPath(self.settingsFileDir, self.currentFileName)
+      newFilePath = os.joinPath(self.settingsFileDir, fileName)
+    self.currentFileName = fileName
+    try:
+      moveFile(currentFilePath, newFilePath)
```
Comment (MishkaRogachev) [inline src/app/global/local_account_settings.nim:42]: may `if currentFilePath != newFilePath`: be useful here?

**Review by MishkaRogachev — APPROVED** [review]
Many thanks! That was an annoying bug

**Review by stefandunca — APPROVED** [review]
🙏

File: src/app/global/local_account_settings.nim (hunk 1 above)
Comment (saledjenic) [reply src/app/global/local_account_settings.nim]: @MishkaRogachev the way how we use it, not needed, cause even if the source and the destination path are the same that doesn't result in an error, but anyhow I added the guard you suggested.

**Comment by saledjenic** [comment]
@status-im/desktop-devs what do you think about two points raised here https://github.com/status-im/status-desktop/pull/13028#issuecomment-1860794166

**Comment by saledjenic** [comment]
@Cuteivist no need for any special id or so, there are already many unique properties for each account, chat key, public key, 3-word name, we can derive an address for some special path that will be used only for that purpose and so on.

**Pull request merged**